x-pack/legacy/plugins/maps/common/descriptor_types.d.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IFieldType } from '../../../../../src/plugins/data/common/index_patterns/fields';
+
+export interface ISourceDescriptor {
+  id: string;
+  type: string;
+}
+
+export interface ILayerDescriptor {
+  sourceDescriptor: ISourceDescriptor;
+  id: string;
+}
+
+export interface IMapSavedObject {
```

Comment:
fwiw, I'd move `IMapSavedObject` to the `maps_telemetry` module. This isn't really a `descriptor` in the way we look at it in Maps (ie. Redux-store state). Most of this is metadata added by the saved-object framework. The "Maps-specific" stuff is embedded in `attributes` anyway.